tests/units/Glpi/Form/AccessControl/FormAccessControlManager.php

```diff
@@ -0,0 +1,552 @@
+                'tech'      => false,
+                'normal'    => false,
+                'post-only' => false,
+            ],
+        ];
+        $_GET['token'] = 'my_token';
+        yield [
+            // Valid token
+            'form'     => $form_3,
+            'expected' => [
+                'glpi'      => true, // Admin can see all forms
+                'tech'      => true,
+                'normal'    => true,
+                'post-only' => true,
+            ],
+        ];
+        unset($_GET['token']);
+
+        // Form with an allow list
+        $form_4 = $this->createForm(
+            (new FormBuilder())
+                ->addAccessControl(AllowList::class, new AllowListConfig([
+                    'user_ids' => [
+                        getItemByTypeName(User::class, "tech", true),
+                    ],
+                    'group_ids' => [
+                        getItemByTypeName(Group::class, "_test_group_1", true),
+                    ],
+                    'profile_ids' => [
+                        getItemByTypeName(Profile::class, "Self-service", true),
+                    ],
+                ]))
+        );
+        yield [
+            'form'     => $form_4,
+            'expected' => [
+                'glpi'      => true, // Admin can see all forms
+                'tech'      => true, // Allowed by user list
+                'normal'    => false,
+                'post-only' => true, // Allowed by profile list
+            ],
+        ];
+
+        // Add "normal" to the allowed group
+        $this->createItem(Group_User::class, [
+            'groups_id' => getItemByTypeName(Group::class, "_test_group_1", true),
+            'users_id'  => getItemByTypeName(User::class, "normal", true),
+        ]);
+        yield [
+            'form'     => $form_4,
+            'expected' => [
+                'glpi'      => true, // Admin can see all forms
+                'tech'      => true, // Allowed by user list
+                'normal'    => true, // Allowed by group list
+                'post-only' => true, // Allowed by profile list
+            ],
+        ];
+
+        // Mix allow list + token usage
+        $form_5 = $this->createForm(
+            (new FormBuilder())
+                ->addAccessControl(AllowList::class, new AllowListConfig([
+                    'user_ids' => [
+                        getItemByTypeName(User::class, "tech", true),
+                    ],
+                ]))
+                ->addAccessControl(DirectAccess::class, new DirectAccessConfig([
+                    'token'               => 'my_token',
+                    'force_direct_access' => true,
+                ]))
+        );
+        yield [
+            'form'     => $form_5,
+            'expected' => [
+                'glpi'      => true, // Admin can see all forms
+                'tech'      => false, // Allowed by user list but token is missing
```

Comment:
IMHO, the access control manager should give access as soon as one of the strategies allows it. I think it is probably how the administrators will expect it to work.

To prevent any ambiguity, it could be possible to declare a configuration at the form level to choose how the strategies results should be evaluated. For instance, the Symfony security component permit do adapt the way results are used when there are multiple ACL "voters", see https://symfony.com/doc/7.1/security/voters.html#changing-the-access-decision-strategy

> `affirmative` (default)
> This grants access as soon as there is one voter granting access;
> `consensus`
> This grants access if there are more voters granting access than denying. In case of a tie the decision is based on the allow_if_equal_granted_denied config option (defaulting to true);
> `unanimous`
> This only grants access if there is no voter denying access.
> `priority`
> This grants or denies access by the first voter that does not abstain, based on their service priority;

In our case, we could limit the option to `affirmative` and `unanimous` cases.

Reply:
I agree to add the affirmative / unanimous" selector, but maybe it could be done in another PR after this one is merged ?
There is already way too much code in this one.

Reply:
I've added it into the backlog: https://github.com/orgs/glpi-project/projects/7/views/1?pane=issue&itemId=60570464

Reply:
OK to do it later.